### PR TITLE
Initial Stage Refactoring

### PR DESCRIPTION
**Pull request by yjskrs**
Replace all AddressBook instances and Person instances with CourseBook and Module respectively. Remove some irrelevant methods.

### CODE REVIEW

File: src/main/java/igrad/model/CourseBook.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+    /**
+     * Removes {@code key} from this {@code CourseBook}.
+     * {@code key} must exist in the course book.
+     */
+    public void removePerson(Module key) {
+        modules.remove(key);
+    }
+
+    //// util methods
```
Comment (nathanaelseen) [inline src/main/java/igrad/model/CourseBook.java:96]: I only know of `//` comments, `////` i'm not too sure, you might want to double check.

File: src/main/java/seedu/address/logic/commands/AddCommand.java [hunk 2]
```diff
@@ -1,7 +1,6 @@
 package seedu.address.logic.commands;
 
 import static java.util.Objects.requireNonNull;
-import static seedu.address.logic.parser.CliSyntax.PREFIX_ADDRESS;
```
Comment (waynewee) [inline src/main/java/seedu/address/logic/commands/AddCommand.java:4]: Think we should just change the properties of an object instead of removing the entire thing. Extra work

Comment (yjskrs) [reply src/main/java/seedu/address/logic/commands/AddCommand.java:4]: oh yeah but it was in the way when trying to change the addressbook instances so i just used what i did in the tutorials haha

Comment (nathanaelseen) [reply src/main/java/seedu/address/logic/commands/AddCommand.java:4]: hmm either way is fine :)

File: src/main/java/igrad/MainApp.java [hunk 3]
```diff
@@ -48,16 +48,16 @@
...
         super.init();
 
         AppParameters appParameters = AppParameters.parse(getParameters());
         config = initConfig(appParameters.getConfigPath());
 
         UserPrefsStorage userPrefsStorage = new JsonUserPrefsStorage(config.getUserPrefsFilePath());
         UserPrefs userPrefs = initPrefs(userPrefsStorage);
-        AddressBookStorage addressBookStorage = new JsonAddressBookStorage(userPrefs.getAddressBookFilePath());
-        storage = new StorageManager(addressBookStorage, userPrefsStorage);
+        CourseBookStorage courseBookStorage = new JsonCourseBookStorage(userPrefs.getCourseBookFilePath());
```
Comment (nathanaelseen) [inline src/main/java/igrad/MainApp.java:59]: May want to change it eventually to `CourseBook` instead of `CourseBookStorage`, but we would probably leave it later, for now its good!

File: src/main/java/igrad/logic/commands/EditCommand.java [hunk 4]
```diff
@@ -1,105 +1,100 @@
...
+import igrad.model.Model;
+import igrad.model.module.Email;
+import igrad.model.module.Module;
+import igrad.model.module.Name;
+import igrad.model.module.Phone;
+import igrad.model.tag.Tag;
 
 /**
- * Edits the details of an existing person in the address book.
+ * Edits the details of an existing module in the course book.
```
Comment (nathanaelseen) [inline src/main/java/igrad/logic/commands/EditCommand.java:27]: There is some inconsistency with full stops `.` at the back of such `constructor` . I know its been there in the original code-base, would try to fix it.  E.g,


_DeleteCommandParser.java_
```
/**
 * Parses input arguments and creates a new DeleteCommand object
 */
public class DeleteCommandParser implements Parser<DeleteCommand> {
...
```

VS

_EditCommand.java (this current file)_
```
/**
 * Edits the details of an existing person in the address book.
 */
public class EditCommand extends Command {
...
```

File: src/main/java/igrad/logic/parser/CourseBookParser.java [hunk 5]
```diff
@@ -1,26 +1,22 @@
-package seedu.address.logic.parser;
-
-import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
-import static seedu.address.commons.core.Messages.MESSAGE_UNKNOWN_COMMAND;
```
Comment (nathanaelseen) [inline src/main/java/igrad/logic/parser/CourseBookParser.java:4]: wait why is this removed  as well? don't we need it?

File: src/main/java/igrad/logic/parser/CourseBookParser.java [hunk 6]
```diff
@@ -56,20 +52,14 @@ public Command parseCommand(String userInput) throws ParseException {
...
-
         case ExitCommand.COMMAND_WORD:
             return new ExitCommand();
 
         case HelpCommand.COMMAND_WORD:
             return new HelpCommand();
 
         default:
-            throw new ParseException(MESSAGE_UNKNOWN_COMMAND);
+            throw new ParseException(Messages.MESSAGE_UNKNOWN_COMMAND);
```
Comment (nathanaelseen) [inline src/main/java/igrad/logic/parser/CourseBookParser.java:62]: oh i see why, you used `Messages.MESSAGE_UNKNOWN_COMMAND`, instead of static import (please refer to comment (directly above this))

guess I'll just change it back to static import

File: src/main/java/igrad/logic/parser/DeleteCommandParser.java [hunk 7]
```diff
@@ -1,10 +1,10 @@
-package seedu.address.logic.parser;
+package igrad.logic.parser;
 
-import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static igrad.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
```
Comment (nathanaelseen) [inline src/main/java/igrad/logic/parser/DeleteCommandParser.java:3]: this is the right way! (preserving and using the static imports for these (static) constants instead)

File: src/main/java/igrad/model/CourseBook.java [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+
+    /**
+     * Creates an CourseBook using the Persons in the {@code toBeCopied}
+     */
+    public CourseBook(ReadOnlyCourseBook toBeCopied) {
+        this();
+        resetData(toBeCopied);
+    }
+
+    //// list overwrite operations
```
Comment (nathanaelseen) [inline src/main/java/igrad/model/CourseBook.java:40]: i'm not sure about these comments tho, `////`, i only know these `//`, but its probably in the original code-base as well, would fix that too.

File: src/main/java/igrad/model/CourseBook.java [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+    /**
+     * Resets the existing data of this {@code CourseBook} with {@code newData}.
+     */
+    public void resetData(ReadOnlyCourseBook newData) {
+        requireNonNull(newData);
+
+        setModules(newData.getModuleList());
+    }
+
+    //// module-level operations
```
Comment (nathanaelseen) [inline src/main/java/igrad/model/CourseBook.java:59]: same as above (directly) comment

File: src/test/data/JsonCourseBookStorageTest/invalidAndValidPersonCourseBook.json [hunk 10]
```diff
@@ -2,12 +2,10 @@
   "persons": [ {
```
Comment (nathanaelseen) [inline src/test/data/JsonCourseBookStorageTest/invalidAndValidPersonCourseBook.json:2]: i think the all `.json` test data  files, should  also  have their files names changed (refactored) to e.g, `invalidAndValidModuleCourseBook.json,` etc, I'll try to change it.

File: src/test/java/igrad/logic/LogicManagerTest.java [hunk 11]
```diff
@@ -59,38 +55,32 @@ public void execute_invalidCommandFormat_throwsParseException() {
...
                 new JsonUserPrefsStorage(temporaryFolder.resolve("ioExceptionUserPrefs.json"));
-        StorageManager storage = new StorageManager(addressBookStorage, userPrefsStorage);
+        StorageManager storage = new StorageManager(courseBookStorage, userPrefsStorage);
         logic = new LogicManager(model, storage);
 
         // Execute add command
-        String addCommand = AddCommand.COMMAND_WORD + NAME_DESC_AMY + PHONE_DESC_AMY + EMAIL_DESC_AMY
-                + ADDRESS_DESC_AMY;
-        Person expectedPerson = new PersonBuilder(AMY).withTags().build();
+        String addCommand = AddCommand.COMMAND_WORD + CommandTestUtil.NAME_DESC_AMY + CommandTestUtil.PHONE_DESC_AMY
```
Comment (nathanaelseen) [inline src/test/java/igrad/logic/LogicManagerTest.java:72]: please refer to above (directly) comment

File: src/test/java/igrad/logic/LogicManagerTest.java [hunk 12]
```diff
@@ -1,14 +1,9 @@
...
 
+import static igrad.commons.core.Messages.MESSAGE_INVALID_MODULE_DISPLAYED_INDEX;
+import static igrad.commons.core.Messages.MESSAGE_UNKNOWN_COMMAND;
+import static igrad.testutil.Assert.assertThrows;
 import static org.junit.jupiter.api.Assertions.assertEquals;
-import static seedu.address.commons.core.Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX;
-import static seedu.address.commons.core.Messages.MESSAGE_UNKNOWN_COMMAND;
-import static seedu.address.logic.commands.CommandTestUtil.ADDRESS_DESC_AMY;
-import static seedu.address.logic.commands.CommandTestUtil.EMAIL_DESC_AMY;
-import static seedu.address.logic.commands.CommandTestUtil.NAME_DESC_AMY;
```
Comment (nathanaelseen) [inline src/test/java/igrad/logic/LogicManagerTest.java:8]: please retain static imports (instead of removing them), they're much cleaner. i'll prob change it back then

File: src/test/java/igrad/logic/commands/ClearCommandTest.java [hunk 13]
```diff
@@ -0,0 +1,32 @@
+package igrad.logic.commands;
+
+import static igrad.logic.commands.CommandTestUtil.assertCommandSuccess;
```
Comment (nathanaelseen) [inline src/test/java/igrad/logic/commands/ClearCommandTest.java:3]: i think you removed the static import `import static seedu.address.testutil.TypicalPersons.getTypicalAddressBook;` as well, please use static imports instead! they're much cleaner.

File: src/test/java/igrad/logic/commands/ClearCommandTest.java [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+    public void execute_emptyCourseBook_success() {
+        Model model = new ModelManager();
+        Model expectedModel = new ModelManager();
+
+        assertCommandSuccess(new ClearCommand(), model, ClearCommand.MESSAGE_SUCCESS, expectedModel);
+    }
+
+    @Test
+    public void execute_nonEmptyCourseBook_success() {
+        Model model = new ModelManager(TypicalPersons.getTypicalCourseBook(), new UserPrefs());
```
Comment (nathanaelseen) [inline src/test/java/igrad/logic/commands/ClearCommandTest.java:25]: you wouldn't have to do this, if you used static imports, as per the original code;
`Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());`

(please refer to comment above)

File: src/test/java/igrad/logic/commands/DeleteCommandTest.java [hunk 15]
```diff
@@ -0,0 +1,108 @@
+package igrad.logic.commands;
+
+import static igrad.logic.commands.CommandTestUtil.assertCommandFailure;
+import static igrad.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static igrad.logic.commands.CommandTestUtil.showPersonAtIndex;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
```
Comment (nathanaelseen) [inline src/test/java/igrad/logic/commands/DeleteCommandTest.java:7]: i think  you removed the last 3 lines of static imports as well, please use static imports, they're much cleaner;

```
import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_PERSON;
import static seedu.address.testutil.TypicalIndexes.INDEX_SECOND_PERSON;
import static seedu.address.testutil.TypicalPersons.getTypicalAddressBook;
```

File: src/test/java/igrad/logic/commands/DeleteCommandTest.java [hunk 16]
```diff
@@ -0,0 +1,108 @@
...
+ * Contains integration tests (interaction with the Model, UndoCommand and RedoCommand) and unit tests for
+ * {@code DeleteCommand}.
+ */
+public class DeleteCommandTest {
+
+    private Model model = new ModelManager(TypicalPersons.getTypicalCourseBook(), new UserPrefs());
+
+    @Test
+    public void execute_validIndexUnfilteredList_success() {
+        Module moduleToDelete = model.getFilteredModuleList().get(TypicalIndexes.INDEX_FIRST_PERSON.getZeroBased());
```
Comment (nathanaelseen) [inline src/test/java/igrad/logic/commands/DeleteCommandTest.java:30]: if you had retained the static import (above), this line of code should look this;
'Module moduleToDelete = model.getFilteredModuleList().get(INDEX_FIRST_PERSON.getZeroBased());'

(please refer to above comment)

File: src/test/java/igrad/logic/commands/EditCommandTest.java [hunk 17]
```diff
@@ -0,0 +1,179 @@
+package igrad.logic.commands;
```
Comment (nathanaelseen) [inline src/test/java/igrad/logic/commands/EditCommandTest.java:1]: This class also has the same issue of static imports, please retain them (as per the original code-base), it makes the code much cleaner, and readable.

File: src/test/java/igrad/logic/commands/EditModuleDescriptorTest.java [hunk 18]
```diff
@@ -0,0 +1,54 @@
+package igrad.logic.commands;
```
Comment (nathanaelseen) [inline src/test/java/igrad/logic/commands/EditModuleDescriptorTest.java:1]: this file too: static imports

File: src/test/java/igrad/logic/parser/AddCommandParserTest.java [hunk 19]
```diff
@@ -0,0 +1,123 @@
+package igrad.logic.parser;
```
Comment (nathanaelseen) [inline src/test/java/igrad/logic/parser/AddCommandParserTest.java:1]: same too, static imports!

This bunch of static imports have been removed!
````
import static seedu.address.logic.commands.CommandTestUtil.ADDRESS_DESC_AMY;
import static seedu.address.logic.commands.CommandTestUtil.ADDRESS_DESC_BOB;
import static seedu.address.logic.commands.CommandTestUtil.EMAIL_DESC_AMY;
import static seedu.address.logic.commands.CommandTestUtil.EMAIL_DESC_BOB;
import static seedu.address.logic.commands.CommandTestUtil.INVALID_ADDRESS_DESC;
import static seedu.address.logic.commands.CommandTestUtil.INVALID_EMAIL_DESC;
import static seedu.address.logic.commands.CommandTestUtil.INVALID_NAME_DESC;
import static seedu.address.logic.commands.CommandTestUtil.INVALID_PHONE_DESC;
import static seedu.address.logic.commands.CommandTestUtil.INVALID_TAG_DESC;
import static seedu.address.logic.commands.CommandTestUtil.NAME_DESC_AMY;
import static seedu.address.logic.commands.CommandTestUtil.NAME_DESC_BOB;
import static seedu.address.logic.commands.CommandTestUtil.PHONE_DESC_AMY;
import static seedu.address.logic.commands.CommandTestUtil.PHONE_DESC_BOB;
import static seedu.address.logic.commands.CommandTestUtil.PREAMBLE_NON_EMPTY;
import static seedu.address.logic.commands.CommandTestUtil.PREAMBLE_WHITESPACE;
import static seedu.address.logic.commands.CommandTestUtil.TAG_DESC_FRIEND;
import static seedu.address.logic.commands.CommandTestUtil.TAG_DESC_HUSBAND;
import static seedu.address.logic.commands.CommandTestUtil.VALID_ADDRESS_BOB;
import static seedu.address.logic.commands.CommandTestUtil.VALID_EMAIL_BOB;
import static seedu.address.logic.commands.CommandTestUtil.VALID_NAME_BOB;
import static seedu.address.logic.commands.CommandTestUtil.VALID_PHONE_BOB;
import static seedu.address.logic.commands.CommandTestUtil.VALID_TAG_FRIEND;
import static seedu.address.logic.commands.CommandTestUtil.VALID_TAG_HUSBAND;
import static seedu.address.logic.parser.CommandParserTestUtil.assertParseFailure;
import static seedu.address.logic.parser.CommandParserTestUtil.assertParseSuccess;
import static seedu.address.testutil.TypicalPersons.AMY;
import static seedu.address.testutil.TypicalPersons.BOB;
````

File: src/test/java/igrad/logic/parser/CourseBookParserTest.java [hunk 20]
```diff
@@ -0,0 +1,82 @@
+package igrad.logic.parser;
+
+import static igrad.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static igrad.commons.core.Messages.MESSAGE_UNKNOWN_COMMAND;
+import static igrad.testutil.Assert.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
```
Comment (nathanaelseen) [inline src/test/java/igrad/logic/parser/CourseBookParserTest.java:7]: same: static imports, in particular, this had been removed;
`import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_PERSON;`

File: src/test/java/igrad/logic/parser/DeleteCommandParserTest.java [hunk 21]
```diff
@@ -1,13 +1,11 @@
-package seedu.address.logic.parser;
```
Comment (nathanaelseen) [inline src/test/java/igrad/logic/parser/DeleteCommandParserTest.java:1]: same, static imports removed

**Review by nathanaelseen — APPROVED** [review]
Hey Yijie, thanks for your work, I can see you've modified a lot of files, but my main concern was probably the removal of static imports.

I'll merge this branch for now, and work on all these minor issues I've found.

File: src/main/java/igrad/model/CourseBook.java (hunk 1 above)
Comment (yjskrs) [reply src/main/java/igrad/model/CourseBook.java:96]: this was from the original code, i didnt change any comments that didnt have anything to do with refactoring

Comment (nathanaelseen) [reply src/main/java/igrad/model/CourseBook.java:96]: okie I know, i'll change it back then!

File: src/main/java/igrad/MainApp.java (hunk 3 above)
Comment (yjskrs) [reply src/main/java/igrad/MainApp.java:59]: oh i used the same format as the original, cos the original ab3 differentiated between 'address book' and 'address book storage', but we can change it as we see fit!

Comment (nathanaelseen) [reply src/main/java/igrad/MainApp.java:59]: yeap, probably not in near future (far future hmm)

File: src/main/java/igrad/logic/commands/EditCommand.java (hunk 4 above)
Comment (yjskrs) [reply src/main/java/igrad/logic/commands/EditCommand.java:27]: Sorry i didn't catch this! Each person can probably just look out for this when they are modifying the original code and then we can have a final check at the end :)

File: src/main/java/igrad/logic/parser/CourseBookParser.java (hunk 5 above)
Comment (yjskrs) [reply src/main/java/igrad/logic/parser/CourseBookParser.java:4]: errr it has been ported to CourseBookParser if I'm not wrong

File: src/main/java/igrad/logic/parser/CourseBookParser.java (hunk 6 above)
Comment (yjskrs) [reply src/main/java/igrad/logic/parser/CourseBookParser.java:62]: Yeah sorry its supposed to be static import.
Some of them changed to non-static when I refactored, I tried my best to fix to static but I guess I missed some, my bad!

File: src/test/data/JsonCourseBookStorageTest/invalidAndValidPersonCourseBook.json (hunk 10 above)
Comment (yjskrs) [reply src/test/data/JsonCourseBookStorageTest/invalidAndValidPersonCourseBook.json:2]: Oh no I missed this sorry!

File: src/test/java/igrad/logic/LogicManagerTest.java (hunk 12 above)
Comment (yjskrs) [reply src/test/java/igrad/logic/LogicManagerTest.java:8]: Yes thank you!
